src/test/ui/issues/issue-28098.rs

```diff
@@ -16,6 +17,7 @@ pub fn other() {
 
     let _ = Iterator::next(&mut ());
     //~^ ERROR `()` is not an iterator
+    //~| ERROR `()` is not an iterator
```

Comment:
This file shows the case where we duplicate the errors. We should be able to elide the error pointing at `Iterator::next` with some judicious use of `Ty::Error`.